src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\Annotation;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class NoSuperfluousPhpdocTagsFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Removes `@param` and `@return` tags that don\'t provide any useful information.',
+            [
+                new CodeSample('<?php
+class Foo {
+    /**
+     * @param Bar $bar
+     */
+    public function doFoo(Bar $bar) {}
+}
+'),
+                new VersionSpecificCodeSample('<?php
+class Foo {
+    /**
+     * @param Bar $bar
+     *
+     * @return Baz
+     */
+    public function doFoo(Bar $bar): Baz {}
+}
+', new VersionSpecification(70000)),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run before NoEmptyPhpdocFixer
```

Comment:
How should this run in relation to `phpdoc_return_self_reference`?

Reply:
Nice catch! Test added and priority updated.